File: knowledge-network-system/backend/app/schemas/graph.py
from pydantic import BaseModel


class GraphBuildRequest(BaseModel):
    text: str
    prefer_llm: bool = True


class GraphNode(BaseModel):
    id: str
    label: str
    type: str = "keyword"
    score: float = 1.0


class GraphEdge(BaseModel):
    source: str
    target: str
    weight: float = 1.0
    relation: str = "co_occurs"


class GraphStats(BaseModel):
    node_count: int
    edge_count: int
    keyword_count: int
    entity_count: int


class GraphBuildResponse(BaseModel):
    nodes: list[GraphNode]
    edges: list[GraphEdge]
    keywords: list[str] = []
    entities: list[str] = []
    stats: GraphStats

File: knowledge-network-system/frontend/src/types/graph.ts
export interface GraphNode {
  id: string;
  label: string;
  type?: string;
  score?: number;
}

export interface GraphEdge {
  source: string;
  target: string;
  weight?: number;
  relation?: string;
}

export interface GraphBuildResponse {
  nodes: GraphNode[];
  edges: GraphEdge[];
  keywords?: string[];
  entities?: string[];
  stats?: {
    node_count: number;
    edge_count: number;
    keyword_count: number;
    entity_count: number;
  };
}
